openapi.yaml

```diff
@@ -0,0 +1,1252 @@
+            encoded as base64. If you are using `Content-Type: application/json`
+            and your payload is using the utf8 character set, then you should
+            use the `utf8` field instead. 
+          examples:
+            - eyJrZXkiOiJ2YWx1ZSJ9
+        utf8:
+          type: string
+          description: >-
+            This is an a convenience field useful for clients that are
+            communicating with Querator via`Content-Type: application/json`.
+            This field assumes a utf8 encoded payload and does not require the
+            payload to be encoded as base64. As such is is an excellent choice
+            for clients who do not have access to a base64 encoder and can't use
+            the `bytes` field for their payload.
+
+
+            NOTE: Both `bytes` and `utf8` can be set, but `bytes` takes
+            precedence. If `bytes` is set, then that will be used as the
+            payload. If `utf8` is set and `bytes` is empty, then `utf8` will be
+            used. If both `bytes` and `utf8` are set, then `bytes` will be used
+            and `utf8` will be dropped.
+          examples:
+            - Hello World!
+            - '{"key", "value"}'
+    QueueReserveItem:
+      type: object
+      properties:
+        id:
+          type: string
+          description: ' A unique id which identifies an item in a queue'
+        encoding:
+          type: string
+          description: ' A user specified field which indicates the encoding the user used to encode the payload'
+        kind:
+          type: string
+          description: >-
+            A Kind or Type the payload contains. Consumers can use this field to
+            determine handling  of the payload prior to unmarshalling
+        reference:
+          type: string
+          description: >-
+            This is a user specified field that can be used by the consumer to
+            determine handling of the queue item without needing to unmarshall
+            the payload.
+        attempts:
+          type: integer
+          description: >-
+            The number of times this item has been deferred or reservation timed
+            out during processing by a consumer
+        reserve_deadline:
+          type: string
+          description: >-
+            The date time that Querator will offer up this item to another
+            consumer. If the consumer reserving this item has not marked it
+            complete. The consumer can use this date to decide if it should
+            finalize it's work if the timeout date has expired.
+        bytes:
+          type: string
+          description: >-
+            The payload of the item as an array of raw bytes with no
+            predetermined character set.
+
+
+            NOTE: If `Content-Type: application/json` is used when communicating
+            with Querator, the 'bytes' field will be encoded as base64. This is
```

Comment:
Curious question: QueueProduceItem specifies that `bytes must be base64 encoded`, implying the client should be the one to base64 encode it. However, here it states the field `will be encoded as base64`, implying Querator does the encoding.

Should this read that the _client_ must base64 encode it before sending the request? Or is Querator expected to do the encoding?

Reply:
nice catch, I'll fix this to make it clear.